bridge/src/bridge/deploy.rs

```diff
@@ -93,12 +104,44 @@ impl<T: Transport + Clone> Future for Deploy<T> {
 							self.app.config.foreign.required_confirmations
 						);
 
-						DeployState::Deploying(home_future.join(foreign_future))
+						DeployState::Deploying {
+							home_data: home_data,
+							foreign_data: foreign_data,
+							future: home_future.join(foreign_future)
+						}
 					},
 					Err(err) => return Err(err.into()),
 				},
-				DeployState::Deploying(ref mut future) => {
+				DeployState::Deploying {
+					ref mut future,
+					ref home_data,
+					ref foreign_data
+				} => {
 					let (home_receipt, foreign_receipt) = try_ready!(future.poll().map_err(ErrorKind::Web3));
+
+					let home_contract_address = home_receipt.contract_address.expect("contract creation receipt must have an address; qed");
+					let foreign_contract_address = foreign_receipt.contract_address.expect("contract creation receipt must have an address; qed");
+
+					write_deployment_info(
+						format!("deployment-home-{}", home_contract_address.to_hex()),
```

Comment:
```
format!("deployment-home-{:x}", home_contract_address)
```

is shorter and does 1 allocation less ;)

Reply:
good to know for the future